src/types/Claims.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import { SplitComponent } from "./Components.sol";
+
+struct BasicTransfer {
+    bytes allocatorSignature; // Authorization from the allocator.
+    uint256 nonce; // A parameter to enforce replay protection, scoped to allocator.
+    uint256 expires; // The time at which the transfer or withdrawal expires.
+    uint256 id; // The token ID of the ERC6909 token to transfer or withdraw.
+    uint256 amount; // The token amount to transfer or withdraw.
+    address recipient; // The recipient of the transfer or withdrawal.
+}
+
+struct SplitTransfer {
+    bytes allocatorSignature; // Authorization from the allocator.
+    uint256 nonce; // A parameter to enforce replay protection, scoped to allocator.
+    uint256 expires; // The time at which the transfer or withdrawal expires.
+    uint256 id; // The token ID of the ERC6909 token to transfer or withdraw.
+    SplitComponent[] recipients; // The recipients and amounts of each transfer.
+}
+
+struct BasicClaim {
+    bytes allocatorSignature; // Authorization from the allocator.
+    bytes sponsorSignature; // Authorization from the sponsor.
+    address sponsor; // The account to source the tokens from.
+    uint256 nonce; // A parameter to enforce replay protection, scoped to allocator.
+    uint256 expires; // The time at which the claim expires.
+    uint256 id; // The token ID of the ERC6909 token to allocate.
+    uint256 allocatedAmount; // The original allocated amount of ERC6909 tokens.
+    address claimant; // The claim recipient; specified by the arbiter.
+    uint256 amount; // The claimed token amount; specified by the arbiter.
+}
+
+struct QualifiedClaim {
+    bytes allocatorSignature; // Authorization from the allocator.
+    bytes sponsorSignature; // Authorization from the sponsor.
+    address sponsor; // The account to source the tokens from.
+    uint256 nonce; // A parameter to enforce replay protection, scoped to allocator.
+    uint256 expires; // The time at which the claim expires.
+    bytes32 qualificationTypehash; // Typehash of the qualification payload.
+    bytes qualificationPayload; // Data used to derive qualification hash.
+    uint256 id; // The token ID of the ERC6909 token to allocate.
+    uint256 allocatedAmount; // The original allocated amount of ERC6909 tokens.
+    address claimant; // The claim recipient; specified by the arbiter.
+    uint256 amount; // The claimed token amount; specified by the arbiter.
+}
+
+struct ClaimWithWitness {
```

Comment:
they have equivalent calldata layouts i think, but memory layout is definitely less compact; main "win" is when comparing across multiple claim endpoints, I'm guessing? when considering a single endpoint in isolation it's kind of a wash

I'm quite bullish on using some kind of custom ABI encoding scheme for V1, though maybe the benefits outweigh the complexity since they're internal calls being performed anyway... 🤔